internal/storage/sql/common/rule.go

```diff
@@ -444,6 +486,14 @@ func (s *Store) CreateDistribution(ctx context.Context, r *flipt.CreateDistribut
 		}
 	)
 
+	err := s.distributionValidationHelper(ctx, r.RuleId, r.VariantId, r.FlagKey, r.NamespaceKey)
```

Comment:
it would prob be less error prone if we just pass in `r` and then have the helper pull off `r.RuleId`, `r.VariantId`, etc, as this signature requires the caller to ensure they are passed in the correct order

Reply:
I think we could abstract them both into a common interface since I think the proto generated types expose `Get` methods for each property, so we should be able to use something like:

```go
interface distributionRequest {
  GetFlagKey() string
  GetRuleId() string
 ...
}
```

Then the `distributionValidationHelper` method could have the signature `distributionValidationHelper(ctx context.Context, r distributionRequest)`. Then call the `GetX` methods instead of the properties directly in the helper

Reply:
@markphelps Yup! That works 👍